ShadowDOM/explainer.md

```diff
@@ -0,0 +1,401 @@
+```js
+<script type="css-module" specifier="/foo.css"> 
+  #content { 
+    color: red; 
+  } 
+</script>
+```
+Given this `<script>` tag, the styles could be applied to a DSD as follows:
+```html
+<my-element> 
+  <template shadowrootmode="open" adoptedstylesheets="/foo.css"> 
+    <!-- ... --> 
+  </template> 
+</my-element>
+```
+
+The shadow root will be created with its `adoptedStyleSheets` containing the `"/foo.css"` CSS module script’s CSSStyleSheet instance. This single `CSSStyleSheet` instance can be shared by any number of shadow roots. 
+
+An inline CSS module script could also be imported in a JavaScript module in the usual way: 
+```html
+import styles from '/foo.css' with { type: 'css' };
+```
+
+An advantage of this approach is that it can be extended to solve similar issues with other content types. Consider the case of a declarative component with many instances stamped out on the page. In the same way that the CSS must either be duplicated in the markup of each component instance or set up using script, the same problem applies to the HTML content of each component. We can envision an inline version of [HTML module scripts](https://github.com/WICG/webcomponents/blob/gh-pages/proposals/html-modules-explainer.md) that would be declared once and applied to any number of shadow root instances: 
+```html
+<template type="module" specifier="/foo.html"> 
+<!-- This template defines an HTML module whose contents are given by the markup 
+     placed here, inserted into the module map with the specifier "/foo.html" --> 
+... 
+</template> 
+<my-element> 
+  <!-- The `shadoowroothtml` attribute causes the `<template>` to populate the shadow root by 
+cloning the contents of the HTML module given by the "/foo.html" specifier, instead of 
+parsing HTML inside the <template>. --> 
+  <template shadowrootmode="open" shadowrootadoptedstylesheets="/foo.css" shadowroothtml="/foo.html"></template> 
+</my-element>
+```
+In this example we’ve leveraged the module system to implement declarative template refs. 
+
+Another advantage of this proposal is that it can allow multiple module specifiers in the `adoptedstylesheets` attribute: 
+```html
+<script type="css-module" specifier="/foo.css">  
+  #content {  
+    color: red;  
+  }  
+</script>  
+
+<script type="css-module" specifier="/bar.css">  
+  #content {  
+    font-family: sans-serif;  
+  }  
+</script>  
+
+<my-element>  
+  <template shadowrootmode="open" adoptedstylesheets="/foo.css, /bar.css">  
+    <!-- ... -->  
+  </template>  
+</my-element>
+```
+
+Lastly, This proposal also lines up with the existing [CSS Module Scripts](https://web.dev/articles/css-module-scripts) for imperative Shadow DOM: 
```

Comment:
Consider omitting this section; I think it's pretty clearly implied by the preceding text describing how this proposal builds on top of CSS modules.